.github/workflows/publish-commit-bottles.yml

```diff
@@ -0,0 +1,61 @@
+name: Publish and commit bottles
+
+on: repository_dispatch
+
+jobs:
+  upload:
+    runs-on: ubuntu-latest
+    container:
+      image: homebrew/brew
+    env:
+      HOMEBREW_DEVELOPER: 1
+      HOMEBREW_FORCE_HOMEBREW_ON_LINUX: 1
+      HOMEBREW_GIT_EMAIL: homebrew-test-bot@lists.sfconservancy.org
+      HOMEBREW_GIT_NAME: BrewTestBot
+      HOMEBREW_NO_ANALYTICS: 1
+      HOMEBREW_NO_AUTO_UPDATE: 1
+    steps:
+      - name: Set up git information
+        uses: actions/github-script@0.8.0
+        id: env
+        with:
+          script: |
+            let user = await github.users.getByUsername({
+              username: context.actor
+            })
+            user.data.email = (user.data.email || user.data.id + "+" + user.data.login + "@users.noreply.github.com")
+            console.log(user.data.email)
+            return user.data
+      - name: Download artifacts
+        uses: Homebrew/actions/download-artifact@master
+        with:
+          github_token: ${{secrets.HOMEBREW_GITHUB_API_TOKEN}}
+          workflow: tests.yml
+          pr: ${{github.event.client_payload.pull_request}}
+          name: bottles
+          path: bottles
+      - name: Upload bottles
+        env:
+          HOMEBREW_BINTRAY_USER: BrewTestBot
+          HOMEBREW_BINTRAY_KEY: ${{ secrets.HOMEBREW_BINTRAY_KEY }}
+        run: |
+          brew update-reset
+          brew test-bot --ci-upload --publish
+      - name: Push bottle commit
+        env:
+          GIT_COMMITTER_NAME: ${{steps.env.outputs.name}}
+          GIT_COMMITTER_EMAIL: ${{steps.env.outputs.email}}
+        run: |
+          cd $(brew --repo ${{github.repository}})
+          git commit --amend --no-edit
```

Comment:
This won't work if there are more than one bottles to upload and that could happen more often in homebrew-core. Only the latest commit will get amended, but earlier will be left untouched. Already bumped into this in linuxbrew-core, but it's not so much an issue there.

Reply:
Can you point me to a pull request / workflow run where this happened?

Reply:
https://github.com/Homebrew/linuxbrew-core/pull/19804

https://github.com/Homebrew/linuxbrew-core/commit/7e16d73c4f1ddc9181c3f352bbc0690d9561059f
https://github.com/Homebrew/linuxbrew-core/commit/2f6af06b2bd5aa94840ec9fc62dfe566a560e6d0

Reply:
Actually wait, I think I see what the issue is. We run `git commit --amend --no-edit` so that we know who ran the dispatch event when we look at the git logs (as the `Committer`).

This is probably something that we can fix with some git magic by looking for commits between `HEAD` and `master` authored by BrewTestBot and then amending each one in turn.

Alternatively we could modify `brew bottle` and `brew test-bot` to do this for us based on an environment variable or passed argument.

Reply:
Looks like `git rebase --force master` with the appropriate environment variables set does the right thing.

Reply:
Will take your word on it :D